modules/core/src/smithy4s/codecs/FieldRenderPredicateCompiler.scala

```diff
@@ -0,0 +1,135 @@
+/*
+ *  Copyright 2021-2025 Disney Streaming
+ *
+ *  Licensed under the Tomorrow Open Source Technology License, Version 1.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *     https://disneystreaming.github.io/TOST-1.0.txt
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package smithy4s.codecs
+
+import smithy4s.schema.Field
+import smithy4s.schema.Schema
+
+trait FieldRenderPredicateCompiler { self =>
+  def compile[A](
```

Comment:
let's change the interface to 

```
def compile(label: String, schema: Schema[A]) : A => Boolean 
```

Reply:
If we do that then we lose access to `isRequired`, `getDefaultValue` and other useful methods defined on the `Field` class. These methods delegate to `Schema` so we can still recreate that but I think that it makes more sense to check `isRequired` on a `field` instance rather than on `schema`. 

wdyt?

Reply:
agree with Kasper, checking `isRequired` on a random schema you were given feels odd

Reply:
Mmm, yeah okay. But at least lose the existential type and do 

```
def compile[S, A](field: Field[S, A]) 
```